### PR TITLE
Code deprecations and test changes

### PR DESCRIPTION
**Pull request by justinta**

### CODE REVIEW

**Comment by cachedout** [comment]
@jtand Please document all of these changes in the carbon release notes and update this PR. Thanks.

**Comment by jtand** [comment]
@cachedout Updated the release notes.

File: doc/topics/releases/carbon.rst [hunk 1]
```diff
@@ -17,3 +17,24 @@ Grains Changes
   .. code-block:: jinja
 
       {% set on_vmware = grains['virtual'].lower() == 'vmware' %}
+
+Deprecations
+============
+- The ``boto_vpc`` execution module had two functions removed,
```
Comment (cachedout) [inline doc/topics/releases/carbon.rst]: With these, let's provide the full name. `boto_vpc.associate_new_dhcp_options_to_vpc` versus the way it's currently phrased. That will make it easier to read and understand, I think.

File: doc/topics/releases/carbon.rst [hunk 2]
```diff
@@ -17,3 +17,24 @@ Grains Changes
...
+  ``associate_new_dhcp_options_to_vpc`` and
+  ``associate_new_network_acl_to_subnet`` in favor of more concise function
+  names, ``create_dhcp_options`` and ``create_network_acl``, respectively.
+
+- The ``data`` execution module had ``getval`` and ``getvals`` functions removed
+  in favor of one function, ``get``, which combines the functionality of the
+  removed functions.
+
+- The ``cache`` runner no longer accpets ``outputter`` or ``minion`` as keyword arguments.
```
Comment (cachedout) [inline doc/topics/releases/carbon.rst]: `cache.grains runner`, not the `cache runner`

File: doc/topics/releases/carbon.rst (hunk 1 above)
Comment (jtand) [inline doc/topics/releases/carbon.rst]: Done!

**Comment by jtand** [comment]
@cachedout Updated.

**Comment by cachedout** [comment]
@jtand Sadly, now we have a merge conflict. (Probably in the release notes.) Could you please rebase this and I'll get it in? Thanks.

**Comment by cachedout** [comment]
Go Go Jenkins!

**Pull request merged**